
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>1. What are the most present protein domains per biological function? &#8212; Notebooks for data science</title>
    
  <link rel="stylesheet" href="_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"TeX": {"Macros": {"N": "\\mathbb{N}", "floor": ["\\lfloor#1\\rfloor", 1], "bmat": ["\\left[\\begin{array}"], "emat": ["\\end{array}\\right]"]}}, "tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="_static/butterfly.png"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="2. What is the contribution of paralogs to SL pairs that share protein domains?" href="paralogs-and-SL-pairs.html" />
    <link rel="prev" title="3. Feature postprocessing from paper." href="prot_domains2lethality-FEATURES-POSTPROCESSING.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="index.html">
  
  <img src="_static/butterfly.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Notebooks for data science</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   To reproduce an example of  Jupyter book, and you are a Windows user:
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Get started
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="test-folder/MysT_Notebooks.html">
   1. Notebooks written entirely in markdown
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="test-folder/python_by_example.html">
   2. An Introductory Example
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="example-notebooks.html">
   3. Jupyter Notebook files
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Machine learning related projects
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="tutorial-categorical-data-on-yeast-interactions.html">
   1. Tutorial on how to handle categorical data from here
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="prot_domains2lethality-PAPER-REPLICATION.html">
   2. Replication of results from paper: “Predicting yeast synthetic lethal genetic interactions using protein domains”
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="prot_domains2lethality-FEATURES-POSTPROCESSING.html">
   3. Feature postprocessing from paper.
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Bioinformatics examples
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   1. What are the most present protein domains per biological function?
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="paralogs-and-SL-pairs.html">
   2. What is the contribution of paralogs to SL pairs that share protein domains?
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        <a class="dropdown-buttons"
            href="_sources/evaluating-protein-domains-per-module.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download notebook file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/evaluating-protein-domains-per-module.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/leilaicruz/jupyter-book"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/leilaicruz/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fevaluating-protein-domains-per-module.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org//v2/gh/leilaicruz/jupyter-book/master?urlpath=lab/tree/mini_book/docs/evaluating-protein-domains-per-module.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#selecting-the-modules-to-evaluate-the-protein-domains">
   1.1. Selecting the modules to evaluate the protein domains
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sorting-the-dataframe-by-the-ratio-of-sharing">
   1.2. Sorting the dataframe by the ratio of sharing
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="what-are-the-most-present-protein-domains-per-biological-function">
<h1><span class="section-number">1. </span>What are the most present protein domains per biological function?<a class="headerlink" href="#what-are-the-most-present-protein-domains-per-biological-function" title="Permalink to this headline">¶</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">intermine.webservice</span> <span class="kn">import</span> <span class="n">Service</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">scipy.io</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span><span class="p">,</span> <span class="n">optimize</span><span class="p">,</span> <span class="n">interpolate</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span> 
<span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">from</span> <span class="nn">scipy.stats</span> <span class="kn">import</span> <span class="n">norm</span><span class="p">,</span> <span class="n">lognorm</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">matplotlib.cm</span> <span class="k">as</span> <span class="nn">cm</span>
<span class="kn">import</span> <span class="nn">matplotlib.mlab</span> <span class="k">as</span> <span class="nn">mlab</span>
<span class="kn">import</span> <span class="nn">os</span><span class="o">,</span> <span class="nn">fnmatch</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="selecting-the-modules-to-evaluate-the-protein-domains">
<h2><span class="section-number">1.1. </span>Selecting the modules to evaluate the protein domains<a class="headerlink" href="#selecting-the-modules-to-evaluate-the-protein-domains" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os</span>
<span class="n">script_dir</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">dirname</span><span class="p">(</span><span class="s1">&#39;__file__&#39;</span><span class="p">)</span> <span class="c1">#&lt;-- absolute dir the script is in</span>
<span class="n">rel_path_domains</span><span class="o">=</span><span class="s2">&quot;datasets/proteins-domains-from-Pfam.xlsx&quot;</span>

<span class="n">abs_file_path_domains</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">script_dir</span><span class="p">,</span> <span class="n">rel_path_domains</span><span class="p">)</span>

<span class="c1"># os.chdir(&#39;../&#39;) #&lt;-- for binder os.chdir(&#39;../&#39;)</span>
<span class="n">my_path_domains</span><span class="o">=</span><span class="n">abs_file_path_domains</span>
<span class="n">data_domains</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">my_path_domains</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="s1">&#39;Unnamed: 0&#39;</span><span class="p">)</span>
<span class="n">data_domains</span><span class="o">=</span><span class="n">data_domains</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data_domains</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>domain-name</th>
      <th>domain-descrip</th>
      <th>domain-start</th>
      <th>domain-end</th>
      <th>domain-method</th>
      <th>domain-id</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>AAC1</td>
      <td>PF00153</td>
      <td>Mito_carr; Mitochondrial substrate/solute carrier</td>
      <td>10</td>
      <td>106</td>
      <td>Pfam</td>
      <td>2056375</td>
    </tr>
    <tr>
      <th>1</th>
      <td>AAC1</td>
      <td>PF00153</td>
      <td>Mito_carr; Mitochondrial substrate/solute carrier</td>
      <td>116</td>
      <td>210</td>
      <td>Pfam</td>
      <td>2056376</td>
    </tr>
    <tr>
      <th>2</th>
      <td>AAC1</td>
      <td>PF00153</td>
      <td>Mito_carr; Mitochondrial substrate/solute carrier</td>
      <td>217</td>
      <td>304</td>
      <td>Pfam</td>
      <td>2056377</td>
    </tr>
    <tr>
      <th>3</th>
      <td>AAC3</td>
      <td>PF00153</td>
      <td>Mito_carr; Mitochondrial substrate/solute carrier</td>
      <td>9</td>
      <td>105</td>
      <td>Pfam</td>
      <td>2062638</td>
    </tr>
    <tr>
      <th>4</th>
      <td>AAC3</td>
      <td>PF00153</td>
      <td>Mito_carr; Mitochondrial substrate/solute carrier</td>
      <td>115</td>
      <td>208</td>
      <td>Pfam</td>
      <td>2062639</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pathways</span><span class="o">=</span><span class="p">[</span> <span class="s1">&#39;galactose degradation&#39;</span><span class="p">,</span><span class="s1">&#39;phospholipid biosynthesis&#39;</span><span class="p">,</span> <span class="s2">&quot;ethanol degradation&quot;</span><span class="p">]</span>
<span class="n">biological_processes_slim</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;cell budding&quot;</span><span class="p">,</span><span class="s2">&quot;lipid binding&quot;</span><span class="p">,</span><span class="s2">&quot;cytokinesis&quot;</span><span class="p">]</span>
<span class="n">biological_processes_goterm</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;cell morphogenesis involved in conjugation with cellular fusion&quot;</span><span class="p">,</span><span class="s2">&quot;budding cell apical bud growth&quot;</span><span class="p">,</span><span class="s2">&quot;establishment of cell polarity&quot;</span><span class="p">,</span><span class="s2">&quot;cytoskeleton organization&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">from_go_to_genes</span><span class="p">(</span><span class="n">go</span><span class="p">,</span><span class="n">label</span><span class="p">):</span>
    <span class="c1">#label=[&quot;GOTerm&quot; or &quot;GOSlimTerm&quot;]</span>
    <span class="n">service</span> <span class="o">=</span> <span class="n">Service</span><span class="p">(</span><span class="s1">&#39;https://yeastmine.yeastgenome.org/yeastmine/service&#39;</span><span class="p">,</span> <span class="n">token</span> <span class="o">=</span> <span class="s1">&#39;K1P7y5Oeb608Hcu06e44&#39;</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">=</span> <span class="n">service</span><span class="o">.</span><span class="n">new_query</span><span class="p">(</span><span class="s2">&quot;Gene&quot;</span><span class="p">)</span>
    <span class="n">query</span><span class="o">.</span><span class="n">add_constraint</span><span class="p">(</span><span class="s2">&quot;goAnnotation.ontologyTerm.parents&quot;</span><span class="p">,</span> <span class="n">label</span><span class="p">)</span>
    <span class="n">query</span><span class="o">.</span><span class="n">add_view</span><span class="p">(</span>
        <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="s2">&quot;goAnnotation.evidence.code.annotType&quot;</span><span class="p">,</span>
        <span class="s2">&quot;goAnnotation.ontologyTerm.parents.name&quot;</span>
    <span class="p">)</span>
    <span class="n">query</span><span class="o">.</span><span class="n">add_constraint</span><span class="p">(</span><span class="s2">&quot;goAnnotation.qualifier&quot;</span><span class="p">,</span> <span class="s2">&quot;!=&quot;</span><span class="p">,</span> <span class="s2">&quot;NOT&quot;</span><span class="p">,</span> <span class="n">code</span> <span class="o">=</span> <span class="s2">&quot;C&quot;</span><span class="p">)</span>
    <span class="n">query</span><span class="o">.</span><span class="n">add_constraint</span><span class="p">(</span><span class="s2">&quot;goAnnotation.qualifier&quot;</span><span class="p">,</span> <span class="s2">&quot;IS NULL&quot;</span><span class="p">,</span> <span class="n">code</span> <span class="o">=</span> <span class="s2">&quot;D&quot;</span><span class="p">)</span>
    <span class="n">query</span><span class="o">.</span><span class="n">add_constraint</span><span class="p">(</span><span class="s2">&quot;goAnnotation.evidence.code.annotType&quot;</span><span class="p">,</span> <span class="s2">&quot;=&quot;</span><span class="p">,</span> <span class="s2">&quot;manually curated&quot;</span><span class="p">,</span> <span class="n">code</span> <span class="o">=</span> <span class="s2">&quot;F&quot;</span><span class="p">)</span>
    <span class="n">query</span><span class="o">.</span><span class="n">add_constraint</span><span class="p">(</span><span class="s2">&quot;goAnnotation.ontologyTerm.parents.name&quot;</span><span class="p">,</span> <span class="s2">&quot;=&quot;</span><span class="p">,</span> <span class="n">go</span><span class="p">,</span> <span class="n">code</span> <span class="o">=</span> <span class="s2">&quot;G&quot;</span><span class="p">)</span>
    <span class="n">query</span><span class="o">.</span><span class="n">set_logic</span><span class="p">(</span><span class="s2">&quot;(C or D) and F and G&quot;</span><span class="p">)</span>

    <span class="n">data_toy</span><span class="o">=</span><span class="n">defaultdict</span><span class="p">(</span><span class="nb">dict</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">row</span><span class="p">,</span><span class="n">counter</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">query</span><span class="o">.</span><span class="n">rows</span><span class="p">(),</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">query</span><span class="o">.</span><span class="n">rows</span><span class="p">()))):</span>

        <span class="n">data_toy</span><span class="p">[</span><span class="s1">&#39;gene-name&#39;</span><span class="p">][</span><span class="n">counter</span><span class="p">]</span><span class="o">=</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;symbol&quot;</span><span class="p">]</span>
        <span class="n">data_toy</span><span class="p">[</span><span class="s1">&#39;evidence&#39;</span><span class="p">][</span><span class="n">counter</span><span class="p">]</span><span class="o">=</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;goAnnotation.evidence.code.annotType&quot;</span><span class="p">]</span>
        <span class="n">data_toy</span><span class="p">[</span><span class="s1">&#39;annotation&#39;</span><span class="p">][</span><span class="n">counter</span><span class="p">]</span><span class="o">=</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;goAnnotation.ontologyTerm.parents.name&quot;</span><span class="p">]</span>


    <span class="n">data_toy_pd</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data_toy</span><span class="p">)</span>
    <span class="n">data</span><span class="o">=</span><span class="n">data_toy_pd</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">()</span>

    <span class="n">data</span><span class="o">.</span><span class="n">index</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="p">))</span>
    <span class="k">return</span> <span class="n">data</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">go</span><span class="o">=</span><span class="n">biological_processes_goterm</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span>
<span class="n">data</span><span class="o">=</span><span class="n">from_go_to_genes</span><span class="p">(</span><span class="n">go</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="s1">&#39;GOTerm&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">big_data</span><span class="o">=</span><span class="p">[]</span>


<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">biological_processes_goterm</span><span class="p">)):</span>
    <span class="n">go</span><span class="o">=</span><span class="n">biological_processes_goterm</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">data</span><span class="o">=</span><span class="n">from_go_to_genes</span><span class="p">(</span><span class="n">go</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="s1">&#39;GOTerm&#39;</span><span class="p">)</span>
    <span class="n">big_data</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;gene-name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
    
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">flat_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">item</span> <span class="k">for</span> <span class="n">sublist</span> <span class="ow">in</span> <span class="n">big_data</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">sublist</span><span class="p">]</span>
<span class="n">res</span> <span class="o">=</span> <span class="p">[</span><span class="n">i</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">flat_list</span> <span class="k">if</span> <span class="n">i</span><span class="p">]</span> 
<span class="n">flat_list_unique</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">(</span><span class="n">res</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Selecting the meaningful columns in the respective dataset</span>
<span class="n">domain_id_list</span><span class="o">=</span><span class="n">data_domains</span><span class="p">[</span><span class="s1">&#39;domain-name&#39;</span><span class="p">]</span>
<span class="c1">#query_gene=data[&#39;gene-name&#39;]</span>
<span class="n">query_gene</span><span class="o">=</span><span class="n">flat_list_unique</span>

<span class="c1"># Initialising the arrays</span>
<span class="n">protein_a_list</span><span class="o">=</span><span class="p">[]</span>

<span class="n">population</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">query_gene</span><span class="p">))</span>


<span class="k">for</span> <span class="n">m</span> <span class="ow">in</span> <span class="n">population</span><span class="p">:</span>

    <span class="n">protein_a</span><span class="o">=</span><span class="n">data_domains</span><span class="p">[</span><span class="n">data_domains</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">query_gene</span><span class="p">[</span><span class="n">m</span><span class="p">]]</span>
    <span class="n">protein_a_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">protein_a</span><span class="p">[</span><span class="s1">&#39;domain-name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
    

    
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">remove_empty_domains</span><span class="p">(</span><span class="n">protein_list_search</span><span class="p">):</span>
    <span class="n">index</span><span class="o">=</span><span class="p">[]</span>
    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">protein_list_search</span><span class="p">)):</span>
        <span class="k">if</span> <span class="n">protein_list_search</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">==</span><span class="p">[]:</span>
            <span class="n">index</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">i</span><span class="p">)</span> <span class="c1">## index of empty values for the protein_a_list meaning they dont have any annotated domain</span>

    <span class="n">y</span><span class="o">=</span><span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">protein_list_search</span><span class="p">))</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">index</span><span class="p">]</span> <span class="c1"># a list with non empty values from protein_a list</span>

    <span class="n">protein_list_search_new</span><span class="o">=</span><span class="p">[]</span>
    
    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">y</span><span class="p">:</span>
        <span class="n">protein_list_search_new</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">protein_list_search</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
        
    <span class="k">return</span> <span class="n">protein_list_search_new</span>

<span class="c1">## evaluating the function</span>

<span class="n">protein_a_list_new</span><span class="o">=</span><span class="n">remove_empty_domains</span><span class="p">(</span><span class="n">protein_a_list</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The empty domain in the data were:&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">protein_a_list</span><span class="p">)</span><span class="o">-</span><span class="nb">len</span><span class="p">(</span><span class="n">protein_a_list_new</span><span class="p">),</span> <span class="s1">&#39;out of&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">protein_a_list</span><span class="p">),</span><span class="s1">&#39;domains&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The empty domain in the data were: 39 out of 287 domains
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">protein_module</span><span class="o">=</span><span class="p">[]</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">protein_a_list_new</span><span class="p">)):</span>
    <span class="n">protein_module</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">(</span><span class="n">protein_a_list_new</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span><span class="c1"># just taking the unique domains per protein in the module </span>
<span class="n">protein_module</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">(</span><span class="n">protein_module</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span> <span class="p">)</span>
<span class="n">unique_domains</span><span class="p">,</span><span class="n">index_domains</span><span class="p">,</span><span class="n">index_counts</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">(</span><span class="n">protein_module</span><span class="p">,</span><span class="n">return_index</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">return_counts</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span> <span class="c1"># counting how many proteins have the same domain per module </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">stats_module</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
<span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;unique-domains&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">unique_domains</span>
<span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;counts-each-domains&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">index_counts</span>
<span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;index-for-original-domains-array&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">index_domains</span>
<span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;ratio-of-sharing&#39;</span><span class="p">]</span><span class="o">=</span><span class="p">(</span><span class="n">index_counts</span><span class="o">-</span><span class="mi">1</span><span class="p">)</span><span class="o">/</span><span class="nb">len</span><span class="p">(</span><span class="n">unique_domains</span><span class="p">)</span>

<span class="n">stats_module</span><span class="o">.</span><span class="n">set_index</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">stats_module</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="sorting-the-dataframe-by-the-ratio-of-sharing">
<h2><span class="section-number">1.2. </span>Sorting the dataframe by the ratio of sharing<a class="headerlink" href="#sorting-the-dataframe-by-the-ratio-of-sharing" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;domain-descrip&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">zeros_like</span><span class="p">(</span><span class="n">unique_domains</span><span class="p">)</span>
<span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;protein-name&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">zeros_like</span><span class="p">(</span><span class="n">unique_domains</span><span class="p">)</span>
<span class="c1">#stats_module[&#39;module-name&#39;]=np.zeros_like(unique_domains)</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">unique_domains</span><span class="p">)):</span>
    <span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;domain-descrip&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">=</span><span class="n">data_domains</span><span class="p">[</span><span class="n">data_domains</span><span class="p">[</span><span class="s1">&#39;domain-name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">unique_domains</span><span class="p">[</span><span class="n">i</span><span class="p">]][</span><span class="s1">&#39;domain-descrip&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()[</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">names</span><span class="o">=</span><span class="n">data_domains</span><span class="p">[</span><span class="n">data_domains</span><span class="p">[</span><span class="s1">&#39;domain-name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">unique_domains</span><span class="p">[</span><span class="n">i</span><span class="p">]][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
    <span class="n">inter</span><span class="o">=</span><span class="nb">set</span><span class="p">(</span><span class="n">names</span><span class="p">)</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">query_gene</span><span class="p">)</span>
    <span class="n">stats_module</span><span class="p">[</span><span class="s1">&#39;protein-name&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">inter</span><span class="p">)</span> <span class="c1"># print the intresection of all proteins with those domains with the protein from the module </span>
    <span class="c1">#stats_module[&#39;module-name&#39;][i]=go</span>
<span class="n">stats_module_sorted</span><span class="o">=</span><span class="n">stats_module</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ratio-of-sharing&#39;</span><span class="p">],</span><span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">stats_module_sorted</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>c:\users\linigodelacruz\appdata\local\continuum\anaconda3\envs\wintest\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  &quot;&quot;&quot;
c:\users\linigodelacruz\appdata\local\continuum\anaconda3\envs\wintest\lib\site-packages\ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>unique-domains</th>
      <th>counts-each-domains</th>
      <th>index-for-original-domains-array</th>
      <th>ratio-of-sharing</th>
      <th>domain-descrip</th>
      <th>protein-name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>8</th>
      <td>PF00069</td>
      <td>20</td>
      <td>18</td>
      <td>0.074510</td>
      <td>Pkinase; Protein kinase domain</td>
      <td>[SPS1, PBS2, ELM1, GIN4, STE20, CLA4, CDC28, S...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>PF00018</td>
      <td>15</td>
      <td>0</td>
      <td>0.054902</td>
      <td>SH3_1; SH3 domain</td>
      <td>[SHO1, YSC84, BOI2, BZZ1, LSB1, MYO5, PIN3, MY...</td>
    </tr>
    <tr>
      <th>17</th>
      <td>PF00169</td>
      <td>8</td>
      <td>39</td>
      <td>0.027451</td>
      <td>PH; Pleckstrin homology domain</td>
      <td>[BEM2, SLM2, BOI2, CLA4, BEM3, BOI1, SLM1, BUD4]</td>
    </tr>
    <tr>
      <th>9</th>
      <td>PF00071</td>
      <td>7</td>
      <td>91</td>
      <td>0.023529</td>
      <td>Ras; Small GTPase superfamily</td>
      <td>[RHO3, CDC42, RSR1, TEM1, RHO2, RHO1, RHO4]</td>
    </tr>
    <tr>
      <th>14</th>
      <td>PF00134</td>
      <td>6</td>
      <td>116</td>
      <td>0.019608</td>
      <td>Cyclin_N; Cyclin, N-terminal</td>
      <td>[CLB4, CLB5, PCL1, CLB1, CLB3, PCL2]</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">stats_module_sorted_diff_zero</span><span class="o">=</span><span class="n">stats_module_sorted</span><span class="p">[</span><span class="n">stats_module_sorted</span><span class="p">[</span><span class="s1">&#39;ratio-of-sharing&#39;</span><span class="p">]</span><span class="o">!=</span><span class="mi">0</span><span class="p">]</span>
<span class="n">plt</span><span class="o">.</span><span class="n">bar</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">stats_module_sorted_diff_zero</span><span class="p">[</span><span class="s1">&#39;domain-descrip&#39;</span><span class="p">],</span><span class="n">height</span><span class="o">=</span><span class="n">stats_module_sorted_diff_zero</span><span class="p">[</span><span class="s1">&#39;ratio-of-sharing&#39;</span><span class="p">]);</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xticks</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;ratio of sharing of domains&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Module:&#39;</span> <span class="o">+</span> <span class="n">go</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0.5, 1.0, &#39;Module:cytoskeleton organization&#39;)
</pre></div>
</div>
<img alt="_images/evaluating-protein-domains-per-module_17_1.png" src="_images/evaluating-protein-domains-per-module_17_1.png" />
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="prot_domains2lethality-FEATURES-POSTPROCESSING.html" title="previous page"><span class="section-number">3. </span>Feature postprocessing from paper.</a>
    <a class='right-next' id="next-link" href="paralogs-and-SL-pairs.html" title="next page"><span class="section-number">2. </span>What is the contribution of paralogs to SL pairs that share protein domains?</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Leila Iñigo<br/>
        
            &copy; Copyright GPL License 3.0.<br/>
          <div class="extra_footer">
            Enjoy coding!
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>