

<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Build a program that reads the paralogs of the que query gene and see if that paralog is also SL of the query gene , by inspecting if the paralog is present in the target genes of the SL database. &#8212; Notebooks for data science</title>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css" integrity="sha384-KA6wR/X5RY4zFAHpv/CnoG2UW1uogYfdnP67Uv7eULvTveboZJg0qUpmJZb5VqzN" crossorigin="anonymous">
    <link href="_static/css/index.css" rel="stylesheet">
    <link rel="stylesheet" href="_static/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/jupyter-sphinx.css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/sphinx-book-theme.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/mystnb.js"></script>
    <script src="_static/sphinx-book-theme.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="https://unpkg.com/@jupyter-widgets/html-manager@^0.18.0/dist/embed-amd.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="What is the protein domain profile of genes from the same module ?" href="evaluating-protein-domains-per-module.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="docsearch:language" content="en">



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="index.html">
  
  <img src="_static/butterfly.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Notebooks for data science</h1>
  
</a>
</div>

<form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>

<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  <p class="caption">
 <span class="caption-text">
  Get started
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="test-folder/MysT_Notebooks.html">
   Notebooks written entirely in markdown
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="test-folder/python_by_example.html">
   An Introductory Example
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Machine learning related projects
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="tutorial-categorical-data-on-yeast-interactions.html">
   Tutorial on how to handle categorical data from here
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="prot_domains2lethality-PAPER-REPLICATION.html">
   Replication of results from paper: “Predicting yeast synthetic lethal genetic interactions using protein domains”
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="prot_domains2lethality-PAPER-REPLICATION.html#building-the-feature-matrix">
   Building the feature matrix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="prot_domains2lethality-PAPER-REPLICATION.html#separate-features-from-labels-to-set-up-the-data-from-the-ml-workflow">
   Separate features from labels to set up the data from the ML workflow
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="prot_domains2lethality-PAPER-REPLICATION.html#using-pca-to-reduce-the-dimensionality-of-the-problem">
   Using PCA to reduce the dimensionality of the problem
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="prot_domains2lethality-FEATURES-POSTPROCESSING.html">
   Feature postprocessing from paper.
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Bioinformatics examples
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="evaluating-protein-domains-per-module.html">
   What is the protein domain profile of genes from the same module ?
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="evaluating-protein-domains-per-module.html#selecting-the-modules-to-evaluate-the-protein-domains">
   Selecting the modules to evaluate the protein domains
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="evaluating-protein-domains-per-module.html#sorting-the-dataframe-by-the-ratio-of-sharing">
   Sorting the dataframe by the ratio of sharing
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Build  a program that reads the paralogs of the que query gene and see if that paralog is also SL of the query gene , by inspecting if the paralog is present in the target genes of the SL database.
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="#putting-1-s-if-the-paralog-pair-is-also-sl">
   Putting 1’s if the paralog pair is also SL
  </a>
 </li>
</ul>

</nav>

 <!-- To handle the deprecated key -->

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        <div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    
    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        <a class="dropdown-buttons"
            href="_sources/paralogs-and-SL-pairs.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download notebook file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/paralogs-and-SL-pairs.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
    
</div>
        <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/leilaicruz/jupyter-book"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/leilaicruz/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fparalogs-and-SL-pairs.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/leilaicruz/jupyter-book/master?urlpath=tree/mini_book/docs/paralogs-and-SL-pairs.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        <button type="button" class="btn btn-secondary topbarbtn"
            onclick="initThebelabSBT()" title="Launch Thebelab" data-toggle="tooltip" data-placement="left"><i
                class="fas fa-play"></i><span style="margin-left: .4em;">Live Code</span></button>
        
    </div>
</div>

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> On this page
        </div>
        <nav id="bd-toc-nav">
            
        </nav>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">scipy.io</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span><span class="p">,</span> <span class="n">optimize</span><span class="p">,</span> <span class="n">interpolate</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span> 
<span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">from</span> <span class="nn">scipy.stats</span> <span class="kn">import</span> <span class="n">norm</span><span class="p">,</span> <span class="n">lognorm</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">matplotlib.cm</span> <span class="k">as</span> <span class="nn">cm</span>
<span class="kn">import</span> <span class="nn">matplotlib.mlab</span> <span class="k">as</span> <span class="nn">mlab</span>
<span class="kn">import</span> <span class="nn">os</span><span class="o">,</span> <span class="nn">fnmatch</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os</span>
<span class="n">script_dir</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">dirname</span><span class="p">(</span><span class="s1">&#39;__file__&#39;</span><span class="p">)</span> <span class="c1">#&lt;-- absolute dir the script is in</span>
<span class="n">rel_path_SL</span> <span class="o">=</span> <span class="s2">&quot;datasets/data-synthetic-lethals.xlsx&quot;</span>

<span class="n">abs_file_path_domains</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">script_dir</span><span class="p">,</span> <span class="n">rel_path_SL</span><span class="p">)</span>

<span class="n">my_path_domains</span><span class="o">=</span><span class="n">abs_file_path_domains</span>
<span class="n">data_domains</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">my_path_domains</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">data_sl</span><span class="o">=</span><span class="n">data_domains</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span>

<span class="n">rel_path_query</span><span class="o">=</span><span class="s2">&quot;datasets/paralogs_sL_from_query-genes.xlsx&quot;</span>
<span class="n">abs_file_path_query</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">script_dir</span><span class="p">,</span> <span class="n">rel_path_query</span><span class="p">)</span>

<span class="n">rel_path_target</span><span class="o">=</span><span class="s2">&quot;datasets/paralogs_sL_from_target-genes.xlsx&quot;</span>
<span class="n">abs_file_path_target</span><span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">script_dir</span><span class="p">,</span> <span class="n">rel_path_target</span><span class="p">)</span>

<span class="n">query_paralogs</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">abs_file_path_query</span><span class="p">)</span>
<span class="n">target_paralogs</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">abs_file_path_target</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="build-a-program-that-reads-the-paralogs-of-the-que-query-gene-and-see-if-that-paralog-is-also-sl-of-the-query-gene-by-inspecting-if-the-paralog-is-present-in-the-target-genes-of-the-sl-database">
<h1>Build  a program that reads the paralogs of the que query gene and see if that paralog is also SL of the query gene , by inspecting if the paralog is present in the target genes of the SL database.<a class="headerlink" href="#build-a-program-that-reads-the-paralogs-of-the-que-query-gene-and-see-if-that-paralog-is-also-sl-of-the-query-gene-by-inspecting-if-the-paralog-is-present-in-the-target-genes-of-the-sl-database" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p>this is the first check to analyze if the reason why a SL pair shares domains is because they are also paralogs.</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># query_paralogs=pd.read_excel(&#39;datasets/paralogs_sL_from_query-genes.xlsx&#39;)</span>
<span class="c1"># target_paralogs=pd.read_excel(&#39;datasets/paralogs_sL_from_target-genes.xlsx&#39;)</span>

<span class="n">query_paralogs_pd</span><span class="o">=</span><span class="n">query_paralogs</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="s1">&#39;Unnamed: 0&#39;</span><span class="p">)</span>
<span class="n">query_paralogs_pd</span><span class="o">.</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">,</span><span class="s1">&#39;name-paralogue&#39;</span><span class="p">]</span>

<span class="n">target_paralogs_pd</span><span class="o">=</span><span class="n">target_paralogs</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="s1">&#39;Unnamed: 0&#39;</span><span class="p">)</span>
<span class="n">target_paralogs_pd</span><span class="o">.</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">,</span><span class="s1">&#39;name-paralogue&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">query_paralogs_pd</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
<span class="n">target_paralogs_pd</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name-gene</th>
      <th>name-paralogue</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>FKH2</td>
      <td>FKH1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>IXR1</td>
      <td>ABF2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>LEM3</td>
      <td>CDC50</td>
    </tr>
    <tr>
      <th>3</th>
      <td>SHC1</td>
      <td>SKT5</td>
    </tr>
    <tr>
      <th>4</th>
      <td>VIK1</td>
      <td>CIK1</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">indexes_sl_query</span><span class="o">=</span><span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">query_paralogs_pd</span><span class="p">)):</span>
    <span class="n">paralog_target</span><span class="o">=</span><span class="n">query_paralogs_pd</span><span class="p">[</span><span class="n">query_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">query_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]][</span><span class="s1">&#39;name-paralogue&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
    <span class="n">list_targets_sl</span><span class="o">=</span><span class="n">data_sl</span><span class="p">[</span><span class="n">data_sl</span><span class="p">[</span><span class="s1">&#39;gene-query-name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">query_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]][</span><span class="s1">&#39;gene-target-name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>


    
    <span class="k">if</span> <span class="n">paralog_target</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">in</span> <span class="n">list_targets_sl</span><span class="p">:</span>
        <span class="n">indexes_sl_query</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">query_paralogs_pd</span><span class="p">[</span><span class="n">query_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-paralogue&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">paralog_target</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>


<span class="n">indexes_sl_target</span><span class="o">=</span><span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">target_paralogs_pd</span><span class="p">)):</span> 
    
    <span class="n">paralog_query</span><span class="o">=</span><span class="n">target_paralogs_pd</span><span class="p">[</span><span class="n">target_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">target_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]][</span><span class="s1">&#39;name-paralogue&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
    <span class="n">list_queries_sl</span><span class="o">=</span><span class="n">data_sl</span><span class="p">[</span><span class="n">data_sl</span><span class="p">[</span><span class="s1">&#39;gene-target-name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">target_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]][</span><span class="s1">&#39;gene-query-name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>


    <span class="k">if</span> <span class="n">paralog_query</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">in</span> <span class="n">list_queries_sl</span><span class="p">:</span>
        <span class="n">indexes_sl_target</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">target_paralogs_pd</span><span class="p">[</span><span class="n">target_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-paralogue&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">paralog_query</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="putting-1-s-if-the-paralog-pair-is-also-sl">
<h1>Putting 1’s if the paralog pair is also SL<a class="headerlink" href="#putting-1-s-if-the-paralog-pair-is-also-sl" title="Permalink to this headline">¶</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sL_values</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">zeros_like</span><span class="p">(</span><span class="n">query_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">])</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">query_paralogs_pd</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">indexes_sl_query</span><span class="p">:</span>
        <span class="n">sL_values</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">=</span><span class="mi">1</span>
<span class="n">query_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;sL&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">sL_values</span>

<span class="n">sL_values</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">zeros_like</span><span class="p">(</span><span class="n">target_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;name-gene&#39;</span><span class="p">])</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">target_paralogs_pd</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">indexes_sl_target</span><span class="p">:</span>
        <span class="n">sL_values</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">=</span><span class="mi">1</span>
<span class="n">target_paralogs_pd</span><span class="p">[</span><span class="s1">&#39;sL&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">sL_values</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">paralogs_sl_pd</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">query_paralogs_pd</span><span class="p">,</span><span class="n">target_paralogs_pd</span><span class="p">],</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sl_that_are_paralogs</span><span class="o">=</span><span class="n">paralogs_sl_pd</span><span class="p">[</span><span class="n">paralogs_sl_pd</span><span class="p">[</span><span class="s1">&#39;sL&#39;</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sl_that_are_paralogs</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">sl_that_are_paralogs</span><span class="p">)))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name-gene</th>
      <th>name-paralogue</th>
      <th>sL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>FKS1</td>
      <td>GSC2</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>RAS2</td>
      <td>RAS1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>GSC2</td>
      <td>FKS1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>KRE6</td>
      <td>SKN1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>DBF2</td>
      <td>DBF20</td>
      <td>1</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>263</th>
      <td>RPS24B</td>
      <td>RPS24A</td>
      <td>1</td>
    </tr>
    <tr>
      <th>264</th>
      <td>RPS26B</td>
      <td>RPS26A</td>
      <td>1</td>
    </tr>
    <tr>
      <th>265</th>
      <td>STD1</td>
      <td>MTH1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>266</th>
      <td>ASK10</td>
      <td>RGC1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>267</th>
      <td>LEU9</td>
      <td>LEU4</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
<p>268 rows × 3 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sl_that_are_paralogs</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-none notranslate"><div class="highlight"><pre><span></span>[&#39;FKS1&#39;, &#39;GSC2&#39;]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">a</span><span class="o">=</span> <span class="p">[</span><span class="s1">&#39;BUD6&#39;</span><span class="p">,</span> <span class="s1">&#39;CDC42&#39;</span><span class="p">]</span>
<span class="n">b</span><span class="o">=</span> <span class="p">[</span><span class="s1">&#39;CDC42&#39;</span><span class="p">,</span> <span class="s1">&#39;BUD6&#39;</span><span class="p">]</span>
<span class="nb">set</span><span class="p">(</span><span class="n">a</span><span class="p">)</span><span class="o">==</span><span class="nb">set</span><span class="p">(</span><span class="n">b</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-none notranslate"><div class="highlight"><pre><span></span>True
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">## use this to compare with the pairs of SL that have shared domains from FEATUREPROCESSINg script  </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pairs_sL</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="s1">&#39;pairs-sL-that-share-domains.npy&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">len</span><span class="p">(</span><span class="n">pairs_sL</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-none notranslate"><div class="highlight"><pre><span></span>748
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">len</span><span class="p">(</span><span class="n">sl_that_are_paralogs</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-none notranslate"><div class="highlight"><pre><span></span>268
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shared_sL_paralogs</span><span class="o">=</span><span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">sl_that_are_paralogs</span><span class="p">)):</span>
    <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">pairs_sL</span><span class="p">)):</span>
        <span class="k">if</span> <span class="nb">set</span><span class="p">(</span><span class="n">sl_that_are_paralogs</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span><span class="o">==</span><span class="nb">set</span><span class="p">(</span><span class="n">pairs_sL</span><span class="p">[</span><span class="n">j</span><span class="p">]):</span>
            <span class="n">shared_sL_paralogs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">pairs_sL</span><span class="p">[</span><span class="n">j</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The contribution of paralogs to the SL pairs that shared domains is =&#39;</span><span class="p">,</span> <span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">shared_sL_paralogs</span><span class="p">)</span><span class="o">/</span><span class="nb">len</span><span class="p">(</span><span class="n">pairs_sL</span><span class="p">),</span><span class="s1">&#39;%&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The contribution of paralogs to the total number of SL pairs is  =&#39;</span><span class="p">,</span> <span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">sl_that_are_paralogs</span><span class="p">)</span><span class="o">/</span><span class="mi">17871</span><span class="p">,</span><span class="s1">&#39;%&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The number of SL that share domains out of the total number of SL pairs is =&#39;</span><span class="p">,</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">pairs_sL</span><span class="p">)</span><span class="o">/</span><span class="mi">17871</span><span class="p">,</span><span class="s1">&#39;%&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-none notranslate"><div class="highlight"><pre><span></span>The contribution of paralogs to the SL pairs that shared domains is = 2.0053475935828877 %
The contribution of paralogs to the total number of SL pairs is  = 1.4996362822449778 %
The number of SL that share domains out of the total number of SL pairs is = 4.185552011638968 %
</pre></div>
</div>
</div>
</div>
</div>


              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="evaluating-protein-domains-per-module.html" title="previous page">What is the protein domain profile of genes from the same module ?</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Leila Iñigo<br/>
        
            &copy; Copyright GPL License 3.0.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    <script src="_static/js/index.js"></script>
    
  </body>
</html>